### ISSUE TITLE
Slow to display tracks for large albums when 'go offline' mode selected

### ISSUE THREAD

**Issue by binhex** [daneren2005/Subsonic#871]
So i have an album with 500 tracks (its a rolling stones best songs ever type thing) and i currently have it starred, which for me means it will cache it to the phones internal SD card, all good so far (confirmed cached to internal sd).

However i have noticed if i put dsub in 'go offline' mode and click on the album it takes an age to show all the tracks, now considering the tracks are cached on the phone you would expect this to be near instantaneous. 

Now comes the weird part, if i switch dsub back to 'go online' then the album listing is fast, which is kinda the opposite of what you would expect.

So in short dsub seems to be slow in displaying tracks in large albums when in offline mode (album cached to internal sdcard), switching to online seems to fix this bug and listing tracks is fast again.

**Comment by arno1979** [comment]
I use Dsub in a similar way than you do and I noticed the same "problem".
Although I wouldn't call it that:

I think this may not really be Dsub's fault: I suspect that while being online the server's database is used to gather the song listing for an album. When switching to the offline mode the SD card's filesystem is used to gather the same info but the access to the filesystem is much slower than the access to the database.
You will hardly notice this when you have an album only containing e.g. 12 tracks or so. But with 500 tracks this will of course be very obvious.

Mind you that I am not a developer and have had no look at Dsub's code but that has always been my explanation for the described behavior.

**Comment by binhex** [comment]
@arno1979 you may very well be right, that does make sense, i wonder if @daneren2005 could confirm this?

**Comment by arno1979** [comment]
@daneren2005 has been very quiet lately... :)

**Comment by daneren2005** [comment]
Yes that is my best guess as well without having the phone physically here to test with. Even reading 500 should be relatively quick unless something is going on with the SD card. I am wondering if it has slow random access or something.

**Comment by binhex** [comment]
OK cool, FYI the card in question is a genuine (tested and verified) Samsung 64GB EVO PLUS Micro SD UHS-I Class 10 Memory Card so i would expect it to perform relatively well, but hey if its just a limitation of reading from card then i can accept that. 

Thanks for chipping in everyone, as there is no action to take i will close the issue.
